com.unity.shadergraph/Editor/ShaderFoundry/Utilities/VirtualTextureAttributes.cs

```diff
@@ -0,0 +1,127 @@
+            if (!string.IsNullOrEmpty(DisplayName))
+                attributeBuilder.Param(new ShaderAttributeParam.Builder(container, DisplayNameParamName, DisplayName).Build());
+            if (!string.IsNullOrEmpty(TextureName))
+                attributeBuilder.Param(new ShaderAttributeParam.Builder(container, TextureNameParamName, TextureName).Build());
+            if (TextureType != LayerTextureType.Default)
+                attributeBuilder.Param(new ShaderAttributeParam.Builder(container, TextureTypeParamName, TextureType.ToString()).Build());
+            return attributeBuilder.Build();
+        }
+
+        internal static VirtualTextureLayerAttribute FindFirst(IEnumerable<ShaderAttribute> attributes)
+        {
+            var attribute = attributes.FindFirst(AttributeName);
+            if (attribute.IsValid)
+                return TryParse(attribute);
+            return null;
+        }
+
+        internal static VirtualTextureLayerAttribute TryParse(ShaderAttribute attribute)
+        {
+            if (attribute.Name != AttributeName)
+                return null;
+            var result = new VirtualTextureLayerAttribute();
+
+            var index = 0;
+            foreach (var param in attribute.Parameters)
+            {
+                if ((index == 0 && string.IsNullOrEmpty(param.Name)) || param.Name == IndexParamName)
+                    int.TryParse(param.Value, out result.Index);
+                else if (param.Name == UniformNameParamName)
+                    result.UniformName = param.Value;
+                else if (param.Name == DisplayNameParamName)
+                    result.DisplayName = param.Value;
+                else if (param.Name == TextureNameParamName)
+                    result.TextureName = param.Value;
+                else if (param.Name == TextureTypeParamName)
+                    Enum.TryParse(param.Value, out result.TextureType);
+                ++index;
+            }
+
+            //var paramDataList = new List<AttributeParsing.ParamData>
+            //{
+            //    new AttributeParsing.ParamData { ParamName = IndexParamName, parseDelegate = (param, data) => { result.Index = int.Parse(param.Value); } },
+            //    new AttributeParsing.ParamData { ParamName = UniformNameParamName, parseDelegate = (param, data) => { result.UniformName = param.Value; } },
+            //    new AttributeParsing.ParamData{ ParamName = DisplayNameParamName, parseDelegate = (param, data) => { result.DisplayName = param.Value; } },
+            //    new AttributeParsing.ParamData{ ParamName = TextureNameParamName, parseDelegate = (param, data) => { result.TextureName = param.Value; } },
+            //    new AttributeParsing.ParamData{ ParamName = TextureTypeParamName, parseDelegate = (param, data) => { Enum.TryParse(param.Value, out result.TextureType); } },
+            //};
+            //AttributeParsing.Parse(attribute, paramDataList);
```

Comment:
Will this get uncommented on the next PR?  If not, can it be removed?  Or is it maybe serving as documentation for something else to come?

Reply:
Whoops, left that in when prototyping a more unified parsing scheme. Will delete.